pkg/pkgmgr/rpm.go

```diff
@@ -599,99 +600,142 @@ func (rm *rpmManager) unpackAndMergeUpdates(ctx context.Context, updates unversi
 			})).Root()
 	}
 
+	// Create a new state for tooling image with all the packages from the image we are trying to patch
+	// this will ensure the rpm database is generate for us to use
+	rpmdb := busyboxCopied.Run(
+		llb.AddEnv("PACKAGES_PRESENT_ALL", string(jsonPackageData)),
+		llb.AddEnv("OS_VERSION", rm.osVersion),
+		llb.Args([]string{
+			`bash`, `-xec`, `
+								json_str=$PACKAGES_PRESENT_ALL
+								packages_formatted=""
+
+								while IFS=':' read -r package version; do
+									pkg_name=$(echo "$package" | sed 's/^"\(.*\)"$/\1/')
+									pkg_version=$(echo "$version" | sed 's/^"\(.*\)"$/\1/')
+
+									packages_formatted="$packages_formatted $pkg_name-$pkg_version"
+
+								done <<< "$(echo "$json_str" | tr -d '{}\n' | tr ',' '\n')"
+
+								tdnf makecache
+								tdnf install -y --releasever=$OS_VERSION --installroot=/tmp/rootfs $packages_formatted
+
+								ls /tmp/rootfs/var/lib/rpm
+						`,
+		})).AddMount("/tmp/rootfs/var/lib/rpm", llb.Scratch())
+
 	// Download all requested update packages without specifying the version. This works around:
 	//  - Reports being slightly out of date, where a newer security revision has displaced the one specified leading to not found errors.
 	//  - Reports not specifying version epochs correct (e.g. bsdutils=2.36.1-8+deb11u1 instead of with epoch as 1:2.36.1-8+dev11u1)
 	//  - Reports specifying remediation packages for cbl-mariner v1 instead of v2 (e.g. *.cm1.aarch64 instead of *.cm2.aarch64)
 	var downloadCmd string
+	const queryFormat = `--queryformat "%s" %s > "%s"`
+
+	pkgs := ""
+
 	if updates != nil {
-		const rpmDownloadTemplate = `yumdownloader --downloadonly --downloaddir=. --best -y %s`
+		rpmDownloadTemplate := `
+		set -x
+		packages="%s"
+		echo "$packages"
+
+		mkdir -p /tmp/rootfs/var/lib
+		ln -s /tmp/rpmdb /tmp/rootfs/var/lib/rpm
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa
+
+		for package in $packages; do
+			package="${package%%.*}" # trim anything after the first "."
+			output=$(tdnf install -y --releasever=$OS_VERSION --installroot=/tmp/rootfs ${package} 2>&1)
+
+			if [ "$IGNORE_ERRORS" = "false" ] && [ $? -ne 0 ]; then
+				exit $?
+			fi
+		done
+
+		mkdir /tmp/rootfs/var/lib/rpmmanifest
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa | tee /tmp/rootfs/var/lib/rpmmanifest/container-manifest-1
+	
+		echo "testing just name"
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa --qf \"\%{NAME}\"
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa
+		rm /tmp/rootfs/var/lib/rpm
+		rm -rf /tmp/rootfs/var/cache/tdnf
+	
+		rpm --dbpath /tmp/rpmdb -qa ` + fmt.Sprintf(queryFormat, resultQueryFormat, pkgs, "/tmp/rootfs/manifest") + ``
+
```

Comment:
looks like this is going to fail validation because we are generating the manifest here, so the expected packages won't match